commands/files/readerfile.go

```diff
@@ -0,0 +1,34 @@
+package files
+
+import "io"
+
+// ReaderFile is a implementation of File created from an `io.Reader`.
+// ReaderFiles are never directories, and can be read from and closed.
+type ReaderFile struct {
+	Filename string
+	Reader   io.Reader
+}
+
+func (f *ReaderFile) IsDirectory() bool {
+	return false
+}
+
+func (f *ReaderFile) NextFile() (File, error) {
+	return nil, ErrNotDirectory
+}
+
+func (f *ReaderFile) FileName() string {
+	return f.Filename
+}
+
+func (f *ReaderFile) Read(p []byte) (int, error) {
+	return f.Reader.Read(p)
+}
+
+func (f *ReaderFile) Close() error {
```

Comment:
It's better to accept a ReadCloser and allow clients to wrap their Readers in a Closer wrapper (that has no effect). 

Secretly introspecting and closing a client's Reader is harder to explain to callers and can result in surprising behavior.